### PR TITLE
Added unit tests for WindowsActionManager class

### PR DESCRIPTION
**Pull request by dtoncu**

### CODE REVIEW

File: argus/unit_tests/action_manager/test_windows.py [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+from argus import util
+
+
+class WindowsActionManagerTest(unittest.TestCase):
+    """Tests for windows action manager class."""
+
+    def setUp(self):
+        self._client = mock.MagicMock()
+        self._config = mock.MagicMock()
+        self._os_type = mock.sentinel
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py]: mock.sentinel.os_type

File: argus/unit_tests/action_manager/test_windows.py [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+        self._action_manager.download(test_utils.URI, test_utils.LOCATION)
+
+        cmd = 'Invoke-WebRequest -Uri {} -OutFile "{}"'.format(
+            test_utils.URI, test_utils.LOCATION)
+        self._client.run_command_with_retry.assert_called_with(
+            cmd, count=util.RETRY_COUNT, delay=util.RETRY_DELAY,
+            command_type=util.POWERSHELL)
+
+    def test_download_exception(self):
+        self._client.run_command_with_retry.side_effect = \
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py]: It is preferred to wrap long lines in parentheses and not a backslash for line continuation.

**Review by stefan-caraiman — COMMENTED** [review]
Are `argus/unit_tests/__init__.py` and `argus/unit_tests/action_manager/__init__.py` required in any way?

File: argus/unit_tests/action_manager/test_windows.py (hunk 2 above)
Comment (dtoncu) [reply argus/unit_tests/action_manager/test_windows.py]: Ok, but here I don't have parentheses :)

**Comment by ader1990** [comment]
looks good to me.

@dtoncu how much is the improvement of coverage?

**Comment by dtoncu** [comment]
The coverage is at 100% (except tests for `install_cbinit` method -- there are in other PR).

File: argus/unit_tests/action_manager/test_windows.py [hunk 3]
```diff
@@ -13,64 +13,818 @@
...
 
     @mock.patch('time.sleep')
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'rmdir')
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'is_dir')
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.rmdir')
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py]: Why are you modifying already existing tests? Please undo these changes.

File: argus/unit_tests/action_manager/test_windows.py [hunk 4]
```diff
@@ -13,64 +13,818 @@
...
+                '.exists')
     def test_git_clone_exists(self, mock_exists):
         mock_exists.return_value = True
-        self.assertRaises(exceptions.ArgusCLIError,
-                          self.action_manager.git_clone,
-                          'fake_url', 'fake_location')
 
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+        with self.assertRaises(exceptions.ArgusCLIError):
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py]: Please undo these changes, this commit has the purpose to add unit tests, not to rewrite already existing ones.

File: argus/unit_tests/action_manager/test_windows.py [hunk 5]
```diff
@@ -13,64 +13,818 @@
...
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_could_not_clone(self, mock_exists):
         mock_exists.return_value = False
-        res = self.action_manager.git_clone('fake_url', 'location', count=0)
-        self.assertFalse(res)
 
-    @mock.patch('argus.action_manager.windows.WindowsActionManager.'
-                'exists')
+        self.assertFalse(self._action_manager.git_clone(
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py]: Ditto.

File: argus/unit_tests/action_manager/test_windows.py [hunk 6]
```diff
@@ -13,64 +13,818 @@
...
+            test_utils.URL, test_utils.LOCATION, count=0))
+
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_successful(self, mock_exists):
         mock_exists.return_value = False
-        self.action_manager._client.run_command.return_value = True
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py:56]: Ditto.

File: argus/unit_tests/action_manager/test_windows.py [hunk 7]
```diff
@@ -13,64 +13,818 @@
...
+                '.rmdir')
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.is_dir')
+    @mock.patch('argus.action_manager.windows.WindowsActionManager'
+                '.exists')
     def test_git_clone_exception(self, mock_exists, mock_is_dir,
                                  mock_rmdir, mock_sleep):
         mock_exists.side_effect = [False, True, True]
         mock_is_dir.return_value = True
-        mock_sleep.return_value = None
```
Comment (stefan-caraiman) [inline argus/unit_tests/action_manager/test_windows.py:71]: Ditto

**Review by stefan-caraiman — CHANGES_REQUESTED** [review]

File: argus/unit_tests/action_manager/test_windows.py (hunk 2 above)
Comment (stefan-caraiman) [reply argus/unit_tests/action_manager/test_windows.py]: You could wrap the whole lines in parentheses, rather than using a backslash.

**Comment by stefan-caraiman** [comment]
Please fix failing tests http://paste.openstack.org/show/rPcU6YJceAfizLp1Eej3/ , and also recheck with pep8/flake8 and pylint, thanks.

File: argus/unit_tests/action_manager/test_windows.py (hunk 4 above)
Comment (dtoncu) [reply argus/unit_tests/action_manager/test_windows.py]: I just wanted to be all test in the same style. :)

File: argus/unit_tests/action_manager/test_windows.py (hunk 7 above)
Comment (dtoncu) [reply argus/unit_tests/action_manager/test_windows.py:71]: In this case, there is no need of this line.

File: argus/unit_tests/action_manager/test_windows.py (hunk 6 above)
Comment (dtoncu) [reply argus/unit_tests/action_manager/test_windows.py:56]: You mocked `run_command` to return `True` value but the real `run_command` method will never return `True`.

**Review by ader1990 — APPROVED** [review]
lgtm

**Pull request merged**